salt/orch/update-etc-hosts.sls

```diff
@@ -1,13 +1,18 @@
+{% if salt['mine.get']('roles:kube-(master|minion)', 'caasp_fqdn', expr_form='grain_pcre').values()|length > 0 %}
 update_pillar:
   salt.function:
-    - tgt: '*'
+    - tgt: 'roles:kube-(master|minion)'
+    - tgt_type: grain_pcre
     - name: saltutil.refresh_pillar
 
 update_grains:
   salt.function:
-    - tgt: '*'
+    - tgt: 'roles:kube-(master|minion)'
```

Comment:
Since we are explicitly refreshing grains and updating the mine on the `kubernetes` orch I guess it's fine, yes. So disregard my comment.